platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/configurationcache/DefaultConfigurationCache.kt

```diff
@@ -565,7 +565,7 @@ class DefaultConfigurationCache internal constructor(
 
     private
     fun registerWatchableBuildDirectories(buildDirs: Iterable<File>) {
-        buildDirs.forEach(virtualFileSystem::registerWatchableHierarchy)
+        buildDirs.forEach { dir -> virtualFileSystem.registerWatchableHierarchy(dir, null) }
```

Comment:
Please advice how to handle this properly.

Reply:
Can you give more context on what you are looking for advice on?

Reply:
See https://github.com/gradle/gradle/blob/ee5dd75f82def82ed632370485f5d75f1509d2c3/platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/configurationcache/DefaultConfigurationCache.kt#L496-L501

There, locations for builds are loaded from the configuration cache and registered in FSW. There is no info about watch probe locations, so I use default values. This should be fine because, IIUC, the registration of such probes happens after a probe for FS is already created and registered, so it's reused.

So, I assume that the current implementation is fine and wanted to double-check that.

Reply:
We should ask the configuration cache team, I'm not aware how this should work.

Reply:
>  I'm not aware how this should work

The configuration cache relies on VFS to compute hashes when it's checking the fingerprint. VFS would refuse to compute hashes for files outside its "watchable directories" thus CC registers all known places before hand.

CC is purely a consumer of the VFS API here.